### ISSUE TITLE
Feature Request: Implement state_dict() / load_state_dict()

### ISSUE THREAD

**Issue by nelson-liu** [JianGoForIt/YellowFin_Pytorch#9]
thanks for this code release, I wish more papers did this! It really makes it effortless to try out this new optimizer :)

It'd be great if YellowFin supported the `state_dict()` and `load_state_dict()` functions, to maintain a consistent serialization API with the other PyTorch optimizers.

**Comment by taey16** [comment]
I'm totally agree with @nelson-liu 's idea.
I think automatic(adaptive) lr and momentum scheduling is much more crucial in large-scale dataset(e.g. imagenet). The imagenet1K dataset needs at least a week to get best accuracy. So, It's very time consuming if we manually tune our parameters. The state_dict() and load_state_dict() operations can reduce our time and efforts dramatically.

Thanks in advance,

**Comment by JianGoForIt** [comment]
@taey16 

Thanks for trying out the optimizer. We have already merged a pull request on this. Please try this feature out and let us know whether there are issues.
